### PR TITLE
i18n(de): add GitHub pages deploy guide

### PR DESCRIPTION
**Pull request by DerYeger**
#### What kind of changes does this PR include?

- Minor content fixes (broken links, typos, etc.)
- Translated content

#### Description

This PR adds a German translation of the guide for deploying Astro sites to GitHub pages.
It also removes a superfluous closing bracket.

### CODE REVIEW

File: src/pages/de/guides/deploy/github.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+Du kannst [GitHub Pages](https://pages.github.com/) verwenden, um eine Astro-Website direkt von einem Repository auf [GitHub.com](https://github.com/) zu hosten.
+
+## Wie du sie veröffentlichst
+
+Du kannst eine Astro-Site bei GitHub Pages veröffentlichen, indem du [GitHub Actions](https://github.com/features/actions) verwendest und so deine Site automatisch erzeugst und veröffentlichst. Um dies zu erreichen, muss dein Quelltext auf GitHub liegen.
+
+Astro pflegt die offizielle `withastro/action`, welche dein Projekt mit minimaler Konfiguration veröffentlicht. Folge den unten angeführten Anweisungen, um deine Astro-Site bei GitHub Pages zu veröffentlichen und les [die Package README](https://github.com/withastro/action), falls du mehr Informationen benötigst.
+
+1. Setze die [`site`](/de/reference/configuration-reference/#site) und, falls benötigt, [`base`](/de/reference/configuration-reference/#base) Optionen in `astro.config.mjs`.
```
Comment (hippotastic) [inline src/pages/de/guides/deploy/github.md:16]:
```suggestion
1. Setze die Optionen [`site`](/de/reference/configuration-reference/#site) und bei Bedarf auch [`base`](/de/reference/configuration-reference/#base) in `astro.config.mjs`.
```

File: src/pages/de/guides/deploy/github.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    - `site` sollte in etwa `https://<DEIN_NUTZERNAME>.github.io` entsprechen
+    - `base` sollte ein Schrägstrich gefolgt vom Namen deines Repositories sein, zum Beispiel `/mein-repository`.
+    
+    :::note
+    Falls dein Repository `<DEIN_NUTZERNAME>.github.io` heißt, benötigst du die Option `base` nicht.
+    :::
+
+2. Erstelle eine neue Datei `.github/workflows/deploy.yml` in deinem Projekt und füge den folgende YAML-Ausschnitt ein.
```
Comment (hippotastic) [inline src/pages/de/guides/deploy/github.md:24]:
```suggestion
2. Erstelle eine neue Datei `.github/workflows/deploy.yml` in deinem Projekt und füge den folgenden YAML-Code ein:
```

File: src/pages/de/guides/deploy/github.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    ```yaml
+    name: Github Pages Astro CI
+
+    on:
+      # Löst den Workflow bei jedem Push auf den `main` Branch aus
```
Comment (hippotastic) [inline src/pages/de/guides/deploy/github.md:30]:
```suggestion
      # Löst den Workflow bei jedem Push auf den `main`-Branch aus
```

File: src/pages/de/guides/deploy/github.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      # Verwendest du einen anderen Branch-Namen? Ersetze `main` durch den Namen deines Branches
+      push:
+        branches: [ main ]
+      # Erlaubt es dir diesen Workflow manuell über den Actions-Tab auf GitHub zu starten
```
Comment (hippotastic) [inline src/pages/de/guides/deploy/github.md:34]:
```suggestion
      # Erlaubt es dir, diesen Workflow manuell über den Actions-Tab auf GitHub zu starten
```

File: src/pages/de/guides/deploy/github.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+      pages: write
+      id-token: write
+
+    jobs:
+      build:
+        runs-on: ubuntu-latest
+        steps:
+          - name: Checke dein Repository mit Git aus
+            uses: actions/checkout@v2          
+          - name: Installiere Abhängigkeiten, erzeuge deine Site und lade sie hoch
```
Comment (hippotastic) [inline src/pages/de/guides/deploy/github.md:49]:
```suggestion
          - name: Installiere Abhängigkeiten, erzeuge deine Website und lade sie hoch
```

File: src/pages/de/guides/deploy/github.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+          name: github-pages
+          url: ${{ steps.deployment.outputs.page_url }}
+        steps:
+          - name: Bei GitHub Pages veröffentlichen
+            id: deployment
+            uses: actions/deploy-pages@v1
+    ```
+    
+    :::caution
+    Die offizielle Astro [Action](https://github.com/withastro/action) such nach einem Lockfile, um deinen bevorzugten Paketmanager (`npm`, `yarn` oder `pnpm`) zu erkennen. Du solltest die von deinem Paketmanager automatisch generierte Datei `package-lock.json`, `yarn.lock` oder `pnpm-lock.yaml` zu deinem Repository hinzufügen.
```
Comment (hippotastic) [inline src/pages/de/guides/deploy/github.md:65]:
```suggestion
    Die offizielle Astro-[Action](https://github.com/withastro/action) sucht nach einem Lockfile, um deinen bevorzugten Paketmanager (`npm`, `yarn` oder `pnpm`) zu erkennen. Du solltest die von deinem Paketmanager automatisch generierte Datei `package-lock.json`, `yarn.lock` oder `pnpm-lock.yaml` zu deinem Repository hinzufügen.
```

File: src/pages/de/guides/deploy/github.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    :::
+
+3. Committe die neue Workflow-Datei und pushe sie zu GitHub.
+
+4. Auf GitHub, gehe zum Tab **Settings** deines Repositories und suche den Abschnitt **Pages**.
```
Comment (hippotastic) [inline src/pages/de/guides/deploy/github.md:70]:
```suggestion
4. Gehe auf GitHub zum Tab **Settings** deines Repositories und suche den Abschnitt **Pages**.  
```

File: src/pages/de/guides/deploy/github.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    
+    :::caution
+    Die offizielle Astro [Action](https://github.com/withastro/action) such nach einem Lockfile, um deinen bevorzugten Paketmanager (`npm`, `yarn` oder `pnpm`) zu erkennen. Du solltest die von deinem Paketmanager automatisch generierte Datei `package-lock.json`, `yarn.lock` oder `pnpm-lock.yaml` zu deinem Repository hinzufügen.
+    :::
+
+3. Committe die neue Workflow-Datei und pushe sie zu GitHub.
+
+4. Auf GitHub, gehe zum Tab **Settings** deines Repositories und suche den Abschnitt **Pages**.  
+
+5. Wähle den `gh-pages` Branch und das `"/" (root)` Verzeichnis als **Source** deiner Site aus und drücke **Save**.
```
Comment (hippotastic) [inline src/pages/de/guides/deploy/github.md:72]:
```suggestion
5. Wähle den `gh-pages`-Branch und das `"/" (root)`-Verzeichnis als **Source** deiner Website aus und drücke **Save**.  
```

File: src/pages/de/guides/deploy/github.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    Die offizielle Astro [Action](https://github.com/withastro/action) such nach einem Lockfile, um deinen bevorzugten Paketmanager (`npm`, `yarn` oder `pnpm`) zu erkennen. Du solltest die von deinem Paketmanager automatisch generierte Datei `package-lock.json`, `yarn.lock` oder `pnpm-lock.yaml` zu deinem Repository hinzufügen.
+    :::
+
+3. Committe die neue Workflow-Datei und pushe sie zu GitHub.
+
+4. Auf GitHub, gehe zum Tab **Settings** deines Repositories und suche den Abschnitt **Pages**.  
+
+5. Wähle den `gh-pages` Branch und das `"/" (root)` Verzeichnis als **Source** deiner Site aus und drücke **Save**.  
+  
+Deine Site sollte jetzt veröffentlicht sein! Wenn du Änderungen an deinem Astro-Projekt zu deinem Repository pushst, wird die GitHub Action diese automatisch veröffentlichen.
```
Comment (hippotastic) [inline src/pages/de/guides/deploy/github.md:74]:
```suggestion
Deine Website sollte jetzt veröffentlicht sein! Wenn du Änderungen an deinem Astro-Projekt zu deinem Repository pushst, wird die GitHub Action diese automatisch veröffentlichen.
```

File: src/pages/de/guides/deploy/github.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+:::tip[Richte eine benutzerdefinierte Domäne ein]
+Optional kannst du eine benutzerdefinierte Domäne einrichten, indem du die folgende Datei `./public/CNAME` zu deinem Projekt hinzufügst:
+
+```txt title="public/CNAME"
+sub.meine-domain.de
+```
+
+Dadurch wird deine Site auf deiner benutzerdefinierten Domäne, statt auf `<DEIN_NUTZERNAME>.github.io`, veröffentlicht. Vergiss nicht, auch [den DNS Eintrag deines Domänen-Anbieters anzupassen](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).
```
Comment (hippotastic) [inline src/pages/de/guides/deploy/github.md:83]:
```suggestion
Dadurch wird deine Website auf deiner benutzerdefinierten Domäne veröffentlicht, und nicht auf `<DEIN_NUTZERNAME>.github.io`. Vergiss nicht, auch [den DNS-Eintrag deines Domänen-Anbieters anzupassen](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).   
```

**Review by hippotastic — COMMENTED** [review]
Really high quality translation! Just had some minor nits. Thank you!

**Comment by DerYeger** [comment]
> Really high quality translation! Just had some minor nits. Thank you!

Thanks for the feedback! I applied your suggestions.

**Pull request merged**